server/sonar-server/src/main/java/org/sonar/server/issue/IssueUpdater.java

```diff
@@ -42,22 +42,30 @@
   private final DbClient dbClient;
   private final IssueStorage issueStorage;
   private final NotificationManager notificationService;
+  private final IssueChangePostProcessor issueChangePostProcessor;
 
-  public IssueUpdater(DbClient dbClient, IssueStorage issueStorage, NotificationManager notificationService) {
+  public IssueUpdater(DbClient dbClient, IssueStorage issueStorage, NotificationManager notificationService,
+    IssueChangePostProcessor issueChangePostProcessor) {
     this.dbClient = dbClient;
     this.issueStorage = issueStorage;
     this.notificationService = notificationService;
+    this.issueChangePostProcessor = issueChangePostProcessor;
   }
 
   /**
    * Same as {@link #saveIssue(DbSession, DefaultIssue, IssueChangeContext, String)} but populates the specified
    * {@link SearchResponseData} with the DTOs (rule and components) retrieved from DB to save the issue.
    */
-  public SearchResponseData saveIssueAndPreloadSearchResponseData(DbSession session, DefaultIssue issue, IssueChangeContext context, @Nullable String comment) {
-    Optional<RuleDefinitionDto> rule = getRuleByKey(session, issue.getRuleKey());
-    ComponentDto project = dbClient.componentDao().selectOrFailByUuid(session, issue.projectUuid());
-    ComponentDto component = dbClient.componentDao().selectOrFailByUuid(session, issue.componentUuid());
-    IssueDto issueDto = saveIssue(session, issue, context, comment, rule, project, component);
+  public SearchResponseData saveIssueAndPreloadSearchResponseData(DbSession dbSession, DefaultIssue issue, IssueChangeContext context,
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Provide multiple methods instead of using "refreshMeasures" to determine which action to take. ([squid:S2301](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2301))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.sonarqube%3Asonarqube&branch=feature%2FSONAR-10117%2Flive-measures&issues=AWB9sCE1qDrCo5ysfgRr&open=AWB9sCE1qDrCo5ysfgRr)

[ISSUE_KEY]: AWB9sCE1qDrCo5ysfgRr